common/os_calls.c

```diff
@@ -3732,3 +3743,37 @@ g_tcp6_bind_address(int sck, const char *port, const char *address)
     return -1;
 #endif
 }
+
+/*****************************************************************************/
+/* returns error, zero is success, non zero is error */
+/* only works in linux */
+int
+g_set_no_new_privs(int only_if_unconfined)
+{
+#if !defined(HAVE_SYS_PRCTL_H) || !defined(PR_SET_NO_NEW_PRIVS)
+    return 0;
+#else
+    /*
+     * If the process is already confined in some way, then AppArmor/SELinux
```

Comment:
Okay. I see that [there is a way for SELinux policies to allow transitions under NNP](https://stackoverflow.com/a/44311446), so hopefully such a problem in the future could be addressed with a policy tweak rather than checking for SELinux confinement here.

(Incidentally, I've seen mention of development of a way of allowing AppArmor transitions under NNP, so perhaps at some point in the future this check could be removed in favor of a slightly more sophisticated AppArmor policy. Not an option for now, however.)